service/frontend/handler.go

```diff
@@ -1362,6 +1362,70 @@ func (wh *WorkflowHandler) SignalWorkflowExecution(ctx context.Context,
 	return nil
 }
 
+// SignalWithStartWorkflowExecution is used to ensure sending a signal event to a workflow execution.
+// If workflow is running, this results in WorkflowExecutionSignaled event recorded in the history
+// and a decision task being created for the execution.
+// If workflow is not running or not found, this results in WorkflowExecutionStarted and WorkflowExecutionSignaled
+// event recorded in history, and a decision task being created for the execution
+func (wh *WorkflowHandler) SignalWithStartWorkflowExecution(ctx context.Context,
+	signalWithStartRequest *gen.SignalWithStartWorkflowExecutionRequest) (*gen.StartWorkflowExecutionResponse, error) {
+	scope := metrics.FrontendSignalWithStartWorkflowExecutionScope
+	sw := wh.startRequestProfile(scope)
+	defer sw.Stop()
+
+	if ok, _ := wh.rateLimiter.TryConsume(1); !ok {
+		return nil, wh.error(createServiceBusyError(), scope)
+	}
+
+	if signalWithStartRequest.Domain == nil {
```

Comment:
do you need to check if Domain == ""?

Reply:
Added. Will check and change all APIs in another request. https://github.com/uber/cadence/issues/581